src/main/java/edu/cornell/kfs/coa/businessobject/SubAccountGlobal.java

```diff
@@ -210,8 +195,24 @@ private A21IndirectCostRecoveryAccount createA21IndirectCostRecoveryAccountFromC
 		icrAccount.setAccountLinePercent(newICR.getAccountLinePercent());
 
 		return icrAccount;
-	}	
+	}
 
+	@Override
+	public void updateGlobalDetailICRAccountCollection(
+			GlobalBusinessObjectDetailBase globalDetail,
+			List<IndirectCostRecoveryAccount> updatedIcrAccounts) {
+		SubAccountGlobalDetail subAccountGlobalDetail = (SubAccountGlobalDetail)globalDetail;
+		List<A21IndirectCostRecoveryAccount> updatedA21IcrAccounts = new ArrayList<A21IndirectCostRecoveryAccount>();
+		for(IndirectCostRecoveryAccount icrAccount : updatedIcrAccounts){
+			updatedA21IcrAccounts.add(A21IndirectCostRecoveryAccount.copyICRAccount(icrAccount));
+		}
+		subAccountGlobalDetail.getSubAccount().getA21SubAccount().setA21IndirectCostRecoveryAccounts(updatedA21IcrAccounts);
+	}
+	
+	public void updateIcrAccounts(GlobalBusinessObjectDetailBase globalDetail, List<IndirectCostRecoveryAccount> icrAccounts){
+		SpringContext.getBean(GlobalObjectWithIndirectCostRecoveryAccountsService.class).updateIcrAccounts(this, globalDetail, icrAccounts);
```

Comment:
Another more SpringContext.getBean() call that could be replaced by a getter/setter.